pxr/imaging/hdSt/material.cpp

```diff
@@ -167,6 +176,72 @@ HdStMaterial::_ProcessTextureDescriptors(
                                        doublesSupported);
 }
 
+#ifdef PXR_MATERIALX_SUPPORT_ENABLED
+void
+HdStMaterial::StartEarlyInit(
+    HdSceneDelegate *sceneDelegate,
+    HdSt_MaterialXSyncDispatcher *syncDispatcher)
+{
+    // The task should be created by this method just once
+    TF_VERIFY(!_filterTask);
+    _filterTask.reset();
+
+    if (!TfGetEnvSetting(HDST_ENABLE_PARALLEL_MTLX_CODEGEN)) {
+        return;
+    }
+
+    HD_TRACE_FUNCTION();
+
+    VtValue vtMat = sceneDelegate->GetMaterialResource(this->GetId());
+    if (!vtMat.IsHolding<HdMaterialNetworkMap>()) {
+        return;
+    }
+
+    HdMaterialNetworkMap const& hdNetworkMap =
+        vtMat.UncheckedGet<HdMaterialNetworkMap>();
+
+    if (hdNetworkMap.terminals.empty() || hdNetworkMap.map.empty()) {
+        return;
+    }
+
+    bool isVolume = false;
+    auto filterTask = std::make_unique<HdSt_MaterialFilterTask>();
```

Comment:
Creating the filter task locally. If some checks fail below, then we don't move it to the Sprim's member and destroy the local.